app/controllers/solutions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SolutionsController < ApplicationController
+  def create
+    @question = Question.find(params[:question_id])
+    @solution = @question.solutions.build(solution_params)
+
+    if @solution.save
+      flash[:notice] = "投稿しました"
+      redirect_back(fallback_location: questions_path)
+    else
+      flash.now[:alert] = "投稿に失敗しました。#{@solution.errors.full_messages}"
+      render "/questions/#{@solution.id}"
```

Comment:
まず`render`の書き方が間違っているようです。
さらに，レンダリングする際の`@solutions`を与え忘れていますよ。

```suggestion
      @solutions = @question.solutions
      render "questions/show"
```